lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -3100,14 +3100,15 @@ void ModuleEmitter::emitRTLModule(rtl::RTLModuleOp module) {
 
       // Append this to the running port decl.
       os << ", " << name;
+      printArraySubscripts(portType, os);
       ++portIdx;
     }
 
     if (portIdx != e)
       os << ',';
-    else
-      os << ");\n";
-    os << '\n';
+    os << "    //" << portType << "\n";
```

Comment:
What do these comments look like with an actual example?  Won't they be super noisy?

Reply:
They don't look noisy to me:

```
module CmpSign(
  input  [3:0] a, b,    //!firrtl.uint<4>
  input  [3:0] c, d,    //!firrtl.sint<4>
  output       out    //!firrtl.flip<uint<1>>
);
```

```
module Print(
  input       clock,    //!firrtl.clock
  input       reset,    //!firrtl.uint<1>
  input [3:0] a, b    //!firrtl.uint<4>
);
```

Reply:
This doesn't look useful, for example, to a Chisel engineer who wants to be able to read pretty verilog, or a customer of an IP.

Reply:
It's kind of important information to have if you need to differentiate between a packed array and an integer. Or a signed int vs an unsigned int. This information is useful for debugging and even more so to a customer for interfacing with the IP produced.

I don't think it makes things less pretty... though maybe less concise.

Reply:
I'd recommend two ways to improve:
1) Only print in the case where it would be ambiguous and therefore helpful.
2) Print in terms the user understands, not in compiler IR terms.

Reply:
Also, it is probably best to split this off this patch since it is a pretty orthogonal improvement.

Reply:
The SystemVerilog type system is so bad I don't see many cases where it wouldn't be ambiguous! But good point about the IR being printed. I'm so used to seeing it that I don't see it.